Misc/NEWS.d/next/Library/2023-05-03-16-50-24.gh-issue-104144.yNkjL8.rst

```diff
@@ -0,0 +1,3 @@
+Optimize asyncio.gather when using eager tasks factory Skip creating
+gathering future and scheduling done callbacks when all futures finish
+without blocking - for up to 3x speedup
```

Comment:
I'm not sure about advertising 3x speedup here; IIUC that's the absolute best-case scenario when _everything_ finishes eagerly, probably not very realistic for most users.

Also the grammar is a little confused here, let's clean that up and maybe add some cross-references.

```suggestion
Optimize :func:`asyncio.gather` when using :func:`asyncio.eager_task_factory`.
Skip creating gathering future and scheduling done callbacks when all futures finish
without blocking.
```